docs/spec_draft.md

```diff
@@ -1406,6 +1407,61 @@ and produces a `result` tensor. More formally,
 ```
 
 [Back to Ops](#index-of-ops)
+
+## stablehlo.rng
+
+### Semantics
+
+Constructs a `result` tensor of a given shape `shape` with random numbers
+generated using the `rng_distribution` algorithm. If `rng_distribution` $=$
+`UNIFORM`, then the random numbers are generated following the uniform
+distribution over the interval [`a`, `b`). If `rng_distribution` $=$ `NORMAL`,
+then the random numbers are generated following the normal distribution with
+mean = `a` and standard deviation = `b`.
+
+If `rng_distribution = UNIFORM` and `b` $\le$ `a`, the result is
+implementation-defined.
+
+### Inputs
+
+| Name               | Type                                                 |
+|--------------------|------------------------------------------------------|
+| `a`                | constant of integer, boolean, or floating-point type |
+| `b`                | constant of integer, boolean, or floating-point type |
+| `shape`            | 1-dimensional tensor of integer type                 |
+| `rng_distribution` | enum of `UNIFORM` and `NORMAL`                       |
+
+### Outputs
+
+| Name     | Type                                               |
+|----------|----------------------------------------------------|
+| `result` | tensor of integer, boolean, or floating-point type |
+
+### Constraints
+
+  * (C1) `a`, `b`, and `result` have the same element type.
+  * (C2) dim(`result`, `i`) = `shape[i]` for all `i` $\in$ [0, size(`shape`)).
+  * (C3) If `rng_distribution = NORMAL`, `a`, `b`, and `result` have the same
+    floating-point element type.
+
+### Examples
+
+```mlir
+// %a = 0
+// %b = 2
+// %shape = [3, 3]
+%result = "stablehlo.rng"(%a, %b, %shape) {
+  rng_distribution = #stablehlo<rng_distribution NORMAL>
+} : (tensor<f32>, tensor<f32>, tensor<2xi32>) -> tensor<3x3xf32>
+// %result: [
+//           [1, 0, 1],
+//           [1, 1, 1],
+//           [0, 0, 0]
+//          ]
```

Comment:
The result doesn't seem to have been updated to use `f32`.

Reply:
Thanks for catching this, I meant to use UNIFORM with `i32`.